pyanaconda/modules/storage/bootloader/grub2.py

```diff
@@ -363,14 +363,23 @@ def write_config(self):
             if rc:
                 log.error("failed to set menu_auto_hide=1")
 
-        # now tell grub2 to generate the main configuration file
-        rc = util.execWithRedirect(
-            "grub2-mkconfig",
-            ["-o", self.config_file],
-            root=conf.target.system_root
-        )
+        # Executes the grub2-common posttrans script to generate grub config files
+        # This should be autohandled by grub2-common, but posttrans scriptlets
+        # in live image scenarios are not triggered.
+        # FIXME: https://bugzilla.redhat.com/show_bug.cgi?id=2327644
+
+        # Drop the first line as it's the posttrans scriptlet header
+        script = util.execWithCapture("rpm", ["-q", "--scripts", "grub2-common"], root=conf.target.system_root).splitlines()[1:]
```

Comment:
This will work only as long as the rootfs deployed by the payload:
- has an RPM database
- has the `rpm` binary
For very minimal systems this might not be the case - eq. the expectation is they will be re-installed instead of updated/modified in place.

Also I'm not sure about immutable system like ostree or bootc/image mode. Eq. if they even have scriptlets like this or if they can/expect changes being done in chroot.

Lastly I wonder about ordering or re-entrancy. Eq. can it depend on some other scritplet being called & can it survive being called more times (or do we need to make sure not to call it twice on regular RPM installs).